installer/scripts/service_control

```diff
@@ -69,49 +176,107 @@ verify_privileges()
+        ps -p `cat $PIDFILE` | grep -q omsagent
+        WS_STATUS=$?
+    else
+        WS_STATUS=$FILE_NOT_FOUND
+    fi
+    return $WS_STATUS
+}
 
+is_omsagent_running()
+{
     # Returns 1 if 'omsagent' is running, 0 otherwise
-    [ -f $PIDFILE ] || return 0
-    ps -p `cat $PIDFILE` | grep -q omsagent
-    STATUS=$?
+    #### Keeping return value on this as it originally was.  I deduce
+    #### that because this value is returned with the is-running qualifier
+    #### to service_control, that there are scripts in other languages
+    #### that use the more conventional values for booleans.  Meanwhile,
+    #### I recommend deprecation in all but the external interface.  XC
+    if this_omsagent_running; then
+        return 1
+    else
+        return 0
+    fi
+} 
+
+remove_PIDfile_unless_omsagent_running()
+{
 
-    # Kill PID file if we're not running any longer
-    if [ $STATUS -ne 0 ]; then
-        rm -f $PIDFILE
+    WS_STATUS=0
+    if this_omsagent_running; then
         return 0
-    else
-        return 1
     fi
+    rm -f $PIDFILE # NOTE:  definition of PIDFILE is dependent on WS.
+    if [ $? -ne 0 ]; then
+        WS_STATUS=2
+        return $WS_STATUS
+    fi
+    return 1
 }
 
 exit_if_agent_not_onboarded()
 {
     if [ ! -f $OMSADMIN_CONF ]; then
-        echo "Warning: Agent is not onboarded. omsagent cannot be registered as a service."
-        exit 0 
+        echo "FATAL: Agent is not onboarded. omsagent cannot be registered as a service."
+        WS_STATUS=$FILE_NOT_FOUND
+        exit $WS_STATUS
     fi
 }
 
+wait_until_omsagent_is_running()
+{
+    # Required parameter: Number of seconds to wait for agent to become running.
+    if [ -z "$1" -o "$1" -le 0 ]; then
+        echo "Function \"wait_until_omsagent_stops\" called with invalid parameter"
+        WS_STATUS=1
+        exit $WS_STATUS
+    fi
+
+    COUNTER=$(( $1 * 2 )) # Since we sleep 0.5 seconds, compute number of seconds
+    while [ $COUNTER -gt 0 ]; do
+        if ! this_omsagent_running; then
+            COUNTER=$(( $COUNTER - 1 ))
+            sleep 0.5
+        else
+            WS_STATUS=0
+            return $WS_STATUS
+        fi
+    done
+
+    # One final try for accurate return status (just return status from the call)
+    remove_PIDfile_unless_omsagent_running
+    WS_STATUS=$? # Should only be non-zero of the removal of a PIDFILE failed for some reason.
+    return $WS_STATUS
+}
+
 wait_until_omsagent_stops()
 {
     # Required parameter: Number of seconds to wait for agent to stop
     if [ -z "$1" -o "$1" -le 0 ]; then
         echo "Function \"wait_until_omsagent_stops\" called with invalid parameter"
-        exit 1
+        WS_STATUS=1
+        exit $WS_STATUS
     fi
 
     COUNTER=$(( $1 * 2 )) # Since we sleep 0.5 seconds, compute number of seconds
     while [ $COUNTER -gt 0 ]; do
-        is_omsagent_running && return $?
-        COUNTER=$(( $COUNTER - 1 ))
-        sleep 0.5
+        if this_omsagent_running; then
```

Comment:
I think you can condense this block: since `remove_PIDfile_unless_omsagent_running` calls `this_omsagent_running` itself, you can just call `remove_PIDfile_unless_omsagent_running`, and sleep if the return value indicates that omsagent is still running.

Reply:
Same.